### PR TITLE
Database for analytics + repos for Delighted

### PR DESCRIPTION
**Pull request by martinraison**
@andrewconner @stkem (@jaredpetker @davoda)

Things to look for:
- Heimdal now has a database
- Created two repos `DelightedUserRepo` and `DelightedAnswerRepo`
- keeping track of user activity ("last_active" in user_values table)
- mobile endpoint for user prefs
- returning "show_delighted_question" field in user prefs (this part may be revised in the future)

Other than that, all the import-related changes occurred when I refactored the code on Heimdal (moving classes between packages)

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/common/time/package.scala [hunk 1]
```diff
@@ -133,12 +133,13 @@ package object time {
...
   lazy val END_OF_TIME = parseStandardTime("9999-01-01 00:00:00.000 -0800")
 
-  implicit class RichDateTime(val date: DateTime) extends AnyVal {
+  implicit class RichDateTime(val date: DateTime) extends AnyVal with Ordered[DateTime] {
     def toLocalDateInZone(implicit zone: DateTimeZone): LocalDate = date.withZone(zone).toLocalDate
     def toLocalTimeInZone(implicit zone: DateTimeZone): LocalTime = date.withZone(zone).toLocalTime
     def toHttpHeaderString: String = HTTP_HEADER_DATETIME_FORMAT.print(date)
     def toStandardTimeString: String = STANDARD_DATETIME_FORMAT.print(date)
     def toStandardDateString: String = STANDARD_DATE_FORMAT.print(date)
+    def compare(that: DateTime): Int = dateTimeOrdering.compare(date, that)
```
Comment (martinraison) [inline kifi-backend/modules/common/app/com/keepit/common/time/package.scala:142]: I'm adding comparison operations to `DateTime`, unless there is a good reason not to do that (or an existing better way)

File: kifi-backend/modules/shoebox/conf/shoebox.routes [hunk 2]
```diff
@@ -132,6 +132,7 @@ POST    /m/1/collections/create             @com.keepit.controllers.mobile.Mobil
 GET     /m/1/collections/all                @com.keepit.controllers.mobile.MobileBookmarksController.allCollections(sort: String ?= "last_kept")
 GET     /m/1/user/me                        @com.keepit.controllers.mobile.MobileUserController.currentUser()
 POST    /m/1/user/me                        @com.keepit.controllers.mobile.MobileUserController.updateCurrentUser()
+GET     /m/1/user/prefs                     @com.keepit.controllers.mobile.MobileUserController.getPrefs()
```
Comment (martinraison) [inline kifi-backend/modules/shoebox/conf/shoebox.routes:136]: @jaredpetker @davoda this is the endpoint you should use to get the user preferences. If the field "show_delighted_question" is set to true in the returned JSON, the question should be shown (other possible values are null and false - ideally should also gracefully handle the case where the field is not there at all).

File: kifi-backend/modules/common/app/com/keepit/common/time/package.scala (hunk 1 above)
Comment (stkem) [inline kifi-backend/modules/common/app/com/keepit/common/time/package.scala:142]: `DateTime` already has a bunch of comparison methods (`isBefore`, `isAfter`, ...). What is the use for this?

Comment (martinraison) [inline kifi-backend/modules/common/app/com/keepit/common/time/package.scala:142]: Oh I didn't know about `isBefore` and `isAfter`. Having `<=`, `>=`, `<`, `>` etc seems even better to me, though (easier to discover, more idiomatic, slightly more readable). I don't really care that much and can revert it if you prefer.

Comment (stkem) [inline kifi-backend/modules/common/app/com/keepit/common/time/package.scala:142]: Nah, it's fine. I was just wondering. `<` and `>` are better.

Comment (martinraison) [inline kifi-backend/modules/common/app/com/keepit/common/time/package.scala:142]: ok, no problem

**Pull request merged**